Packs/HashiCorp-Vault/Integrations/HashiCorpVault/HashiCorpVault.py

```diff
@@ -723,6 +736,58 @@ def get_ch_secrets(engine_path, concat_username_to_cred_name=False):
     return secrets
 
 
+def get_aws_secrets(engine_path, ttl, concat_username_to_cred_name):
+    secrets = []
+    roles_list_url = engine_path + '/roles?list=true'
+    demisto.info('roles_list_url: {}'.format(roles_list_url))
+    res = send_request(roles_list_url, 'get')
+    if not res or 'data' not in res:
+        return []
+    for role in res['data'].get('keys', []):
+        integration_context = demisto.getIntegrationContext()
+        now = datetime.now()
+        if f'{role}_ttl' in integration_context:
+            last = datetime.fromtimestamp(integ_context[f'{role}_ttl'])
+            diff = (now - last).seconds
+            if diff <= ttl - OVERLAP_TIME:
+                continue
+        integ_context[f'{role}_ttl'] = now.timestamp()
+        demisto.setIntegrationContext(integ_context)
```

Comment:
Please use `set_last_run()`

Reply:
I don't think I have last run in the fetch credentials

Reply:
Or did you mean set_integration_context ?